ord_schema/scripts/migration_v0d1d1_v0d2d2.py

```diff
@@ -0,0 +1,359 @@
+            can be serialized to text and parsed as a message within the
+            repeated_new RepeatedCompositeContainer.
+    """
+    for message_old in repeated_old:
+        migrate_CopyFrom(repeated_new.add(), message_old)
+
+
+def migrate_CopyFrom(message_new, message_old):
+    """Mimics the "CopyFrom" method of a message field when the field
+    types are defined equivalently but are different classes. Roundtrips each
+    message through a serialized pbtxt format. Modifies message_new in-place.
+
+    Args:
+        message_new: A message to write the value of message_old into.
+        message_old: A message that can be serialized to text and parsed as a
+            message with the same type as message_new.
+    """
+    serialized = text_format.MessageToString(message_old)
+    text_format.Parse(serialized, message_new)
+
+
+def migrate_MergeFrom_multiple(message_new, message_old, fields):
+    """Iterates over submessage names in "fields" to repeated call
+    migrate_MergeFrom.
+
+    Args:
+        message_new: A message containing at least one repeated field.
+        message_old: A message containing at least one repeated field.
+        fields: A list of field names corresponding to repeated fields that
+            should be merged from the old message to the new one.
+    """
+    for field in fields:
+        migrate_MergeFrom(getattr(message_new, field),
+                          getattr(message_old, field))
+
+
+def migrate_CopyFrom_multiple(message_new, message_old, fields):
+    """Iterates over submessage names in "fields" to repeated call
+    migrate_CopyFrom.
+
+    Args:
+        message_new: A message containing at least one submessage field.
+        message_old: A message containing at least one submessage field.
+        fields: A list of field names corresponding to submessage fields that
+            should be copied from the old message to the new one.
+    """
+    for field in fields:
+        if message_old.HasField(field):
+            migrate_CopyFrom(getattr(message_new, field),
+                             getattr(message_old, field))
+
+
+def migrate_assign_multiple(message_new, message_old, fields):
+    """Iterates over field names in "fields" to assign the values from the
+     old message to the new one.
+
+    Args:
+        message_new: A message.
+        message_old: A message.
+        fields: A list of field names corresponding to fields whose values
+            should be copied from the old message to the new one.
+    """
+    for field in fields:
+        setattr(message_new, field, getattr(message_old, field))
+
+
+def migrate_input(input_new, input_old):
+    """Copies the contents from an old reaction input to the new reaction
+    input format.
+
+    Args:
+        input_new: A reaction_pb2.ReactionInput message.
+        input_old: A reaction_old_pb2.ReactionInput message.
+    """
+    migrate_MergeFrom_multiple(input_new, input_old, ('crude_components',))
```

Comment:
Correct; I suppose this would have raised an error if any of the data had contained a defined crude component amount. Will replace with a NotImplementedError